lms/djangoapps/grades/models.py

```diff
@@ -243,6 +244,7 @@ def save_grade(cls, **kwargs):
         except IntegrityError:
             is_created = False
         if not is_created:
+            grade = cls.objects.get(user_id=user_id, course_id=usage_key.course_key, usage_key=usage_key)
```

Comment:
More correct would be to replace this line and the next with `PersistentGradeWhatever.update_grade()`.  That's precisely what the classmethod exists for.